### PR TITLE
[release/3.0] Fix WriteLargeKeyOrValue System.Text.Json test

### LINKED ISSUE #39856: Utf8JsonWriterTests.WriteLargeKeyOrValue tests are failing
```
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: True, skipValidation: True) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: True, skipValidation: False) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: False, skipValidation: True) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
        Assert.Equal() Failure
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: False, skipValidation: False) [FAIL]
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
```
cc: @ahsonkhan

### PR DESCRIPTION
**Pull request by stephentoub**
Port to release/3.0 of test-only change #39857

When the Utf8JsonWriter is disposed in the first part of the test, it flushes the WriteStartObject token to the ArrayBufferWriter, such that the output.WrittenCount is already 1 when the second part of the test starts, and then the Assert.Equal(0, output.WrittenCount) is doomed to fail.

Fixes #39856

This was just broken by #39560 in master and #39850 in release/3.0.

cc: @ahsonkhan

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]

**Comment by wtgodbe** [comment]
Please hold off on merging this until we open back up for preview9

**Comment by danmosemsft** [comment]
approved (when branch opens)

**Comment by stephentoub** [comment]
> Please hold off on merging this until we open back up for preview9

Ok.  Note, though, that this is fixing a (outer loop) test that's already in release/3.0 that deterministically fails, always.

**Comment by wtgodbe** [comment]
Branch is open, merging

**Pull request merged**
